packages/jest-runtime/src/index.ts

```diff
@@ -898,64 +899,75 @@ class Runtime {
     }
 
     if (moduleName === 'module') {
-      const createRequire = (modulePath: string | URL) => {
-        const filename =
-          typeof modulePath === 'string'
-            ? modulePath.startsWith('file:///')
-              ? fileURLToPath(new URL(modulePath))
-              : modulePath
-            : fileURLToPath(modulePath);
-
-        if (!path.isAbsolute(filename)) {
+      return this._getMockedNativeModule();
+    }
+
+    return require(moduleName);
+  }
+
+  private _getMockedNativeModule(): typeof nativeModule.Module {
+    if (this._moduleImplementation) {
```

Comment:
poor man's cache. might be better to stick in in the module cache we have already